ldms/src/sampler/llnl/lustre_client.c

```diff
@@ -0,0 +1,641 @@
+        "create",
+        "link",
+        "unlink",
+        "symlink",
+        "mkdir",
+        "rmdir",
+        "mknod",
+        "rename",
+        "statfs",
+        "alloc_inode",
+        "setxattr",
+        "getxattr",
+        "getxattr_hits",
+        "listxattr",
+        "removexattr",
+        "inode_permission",
+        NULL
+};
+
+/*******************************************************************/
+/* ldms_metrict_by_name() employs a linear lookup and is verboten for
+ * use by plugins in the main sample() loop.  This section section
+ * implemants a faster lookup using the rbt structure, and offers
+ * alt_ldms_metric_by_name() as a limited alternative to the original.
+ * Someday, perhaps ldms_metric_by_name() will be improved and this
+ * section can be removed.
+ */
+/* red-black tree root for metric index lookup table */
+static struct rbt metric_lookup_tree;
+struct metric_lookup_entry {
+        struct rbn node;
+        int index;
+};
+
+static int metric_lookup_tree_insert(const char *name, int index)
+{
+        struct metric_lookup_entry *entry;
+
+        entry = malloc(sizeof(struct metric_lookup_entry));
+        if (entry == NULL) {
+                return ENOMEM;
+        }
+        entry->index = index;
+        rbn_init(&entry->node, (char *)name);
+        rbt_ins(&metric_lookup_tree, &entry->node);
+
+        return 0;
+}
+
+static void metric_lookup_tree_destroy()
+{
+        struct rbn *rbn;
+        struct metric_lookup_entry *entry;
+
+        while (!rbt_empty(&metric_lookup_tree)) {
+                rbn = rbt_min(&metric_lookup_tree);
+                entry = container_of(rbn, struct metric_lookup_entry, node);
+                rbt_del(&metric_lookup_tree, rbn);
+                free(entry);
+        }
+}
+
+/* "set" is unused in this function.  It is just there to allow this
+   call to be a direct replacement for ldms_metric_by_name().  This plugin
+   only uses one schema, so there is only one global lookup table. */ 
+static int alt_ldms_metric_by_name(ldms_set_t set, const char *name)
+{
+        struct metric_lookup_entry *entry;
+        struct rbn *rbn;
+
+        rbn = rbt_find(&metric_lookup_tree, name);
+        if (rbn == NULL) {
+                return -1;
+        }
+        entry = container_of(rbn, struct metric_lookup_entry, node);
+
+        return entry->index;
+}
+/*******************************************************************/
+
+static int llite_general_schema_is_initialized()
+{
+        if (llite_general_schema != NULL)
+                return 0;
+        else
+                return -1;
+}
+
+static int llite_general_schema_init(const char *schema_name)
+{
+        ldms_schema_t sch;
+        int rc;
+        int i;
+
+        log_fn(LDMSD_LDEBUG, SAMP" llite_general_schema_init()\n");
+        sch = ldms_schema_new(schema_name);
+        if (sch == NULL)
+                goto err1;
+        rc = ldms_schema_meta_array_add(sch, "hostname", LDMS_V_CHAR_ARRAY, 64);
```

Comment:
64 here should be replaced with the appropriate c macro defining the maximum host name limit. HOST_NAME_MAX is probably what you want.

Reply:
Done.